libpromises/policy.c

```diff
@@ -388,6 +408,38 @@ Policy *PolicyMerge(Policy *a, Policy *b)
         bdp->parent_policy = result;
     }
 
+    StringMap *extra_hashes = NULL;
+    if (a->policy_files_hashes != NULL)
+    {
+        result->policy_files_hashes = a->policy_files_hashes;
+        a->policy_files_hashes = NULL;
+        extra_hashes = b->policy_files_hashes;
+    }
+    else if (b->policy_files_hashes != NULL)
+    {
+        result->policy_files_hashes = b->policy_files_hashes;
+        b->policy_files_hashes = NULL;
+    }
+    else
+    {
+        result->policy_files_hashes = NULL;
+    }
+
+    if (extra_hashes != NULL)
+    {
+        MapIterator it = MapIteratorInit((Map*) b->policy_files_hashes);
```

Comment:
Makes more sense to use `extra_hashes` instead of `b->policy_file_hashes` here.

Reply:
Good catch!